sdks/python/apache_beam/transforms/core.py

```diff
@@ -1894,3 +1895,26 @@ def _create_source(serialized_values, coder):
     from apache_beam.transforms.create_source import _CreateSource
 
     return _CreateSource(serialized_values, coder)
+
+
+class Impulse(PTransform):
+    """Impulse primitive."""
+
+    def expand(self, pbegin):
+        if not isinstance(pbegin, pvalue.PBegin):
+          raise TypeError(
+            'Input to Impulse transform must be a PBegin but found %s' % pbegin)
+        return pvalue.PCollection(pbegin.pipeline)
+
+    def get_windowing(self, inputs):
+        return Windowing(GlobalWindows())
+
+    def infer_output_type(self, unused_input_type):
+        return type(None)
```

Comment:
In java Impulse is of type ByteArray Should we keep it the same here.
https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Impulse.java

Reply:
yes, we should.

Reply:
Why? I'd have thought Void would be the obvious choice here. (Changed to bytes for consistency.)

Reply:
you can change it to void but need to follow up so that Dataflow honors this.